Library/Homebrew/cmd/update-report.rb

```diff
@@ -104,6 +105,36 @@ def update_report
       next unless tap.git?
       next if tap.core_tap? && ENV["HOMEBREW_INSTALL_FROM_API"].present? && args.preinstall?
 
+      # HOMEBREW_CORE_MERGE_MAINTAINER can be set by the maintainers
+      # who merge homebrew-core into linuxbrew-core to avoid being migrated
+      # We only migrate HOMEBREW_DEVELOPER users right now
+      # Don't migrate on CI either
+      if ENV["HOMEBREW_DEVELOPER"].present? &&
+         ENV["HOMEBREW_CORE_MERGE_MAINTAINER"].blank? &&
+         ENV["GITHUB_ACTIONS"].blank? &&
+         tap.core_tap? &&
+         tap.remote == "https://github.com/Homebrew/linuxbrew-core"
+        puts_stdout_or_stderr "Migrate from linuxbrew-core to homebrew-core" unless args.quiet?
+
+        LINUXBREW_CORE_MIGRATION_LIST.each do |name|
+          formula = Formula[name]
+          next unless formula.any_version_installed?
+
+          selected_keg = nil
+          version = formula.installed_kegs.map(&:version).max
+          formula.installed_kegs.each do |keg|
+            selected_keg = keg if version == keg.version
+          end
+          tab = Tab.for_keg(selected_keg)
+          tab.version_scheme = tab.version_scheme - 1
+          tab.write
+        end
+
+        safe_system "git", "-C", tap.path, "remote", "set-url", "origin", "https://github.com/Homebrew/homebrew-core"
+        safe_system "git", "-C", tap.path, "fetch", "origin", "master"
```

Comment:
Probably want to allow errors here, to be honest, given that we've already done a homebrew-core fetch.